docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,595 @@
+    item_id: int = items_ids[item_name]
+    grocery_list[item_id].quantity += quantity
+    ```
+    And add the following, to convert the `item_id` to an `int`, and then to increment the quantity of the item by calling the `hincrby` method from redis:
+
+    ```python
+    item_id = int(item_id_str)
+    redis_client.hincrby(f"item_id:{item_id}", "quantity", quantity)
+    ```
+
+We now only need to replace the code for when the item does not exist, i.e. `item_id_str` is `None`. In this case, we will generate a new `item_id`, create a new redis hash for the item, and then add the provided item names and quantity.
+To generate a new `item_id`, let’s use the `incr` method from redis, passing a new hash "item_ids". When this method is run for the first time, it will create the item_ids hash with a unique number, and then each time it's run it will generate an incremental number and store it in this hash.
+
+8. Delete the line with the following content:
+    ```python
+    item_id: int = max(grocery_list.keys()) + 1 if grocery_list else 0
+    ```
+    And add the following:
+    ```python
+    item_id: int = redis_client.incr("item_ids")
+    ```
+
+Now we will add the item to the redis hash, using the `hset` method and by providing a mapping of fields or keys (`item_id`, `quantity` and `name`), and the values (the item’s newly created id, and its provided name and quantity).
+
+9.	Delete the line with the following content:
+    ```python
+    grocery_list[item_id] = ItemPayload(
+            item_id=item_id, item_name=item_name, quantity=quantity
+        )
+    ```
+    And replace it with the following:
+    ```python
+    redis_client.hset(
+                f"item_id:{item_id}",
+                mapping={
+                    "item_id": item_id,
+                    "item_name": item_name,
+                    "quantity": quantity,
+                })
+    ```
+
+Now we only need to map the newly created ID to the item name by setting the hash we referenced in the beginning, `item_name_to_id`.
+
+10.	Add this line to the end of the route, inside the `else` block:
+    ```python
+    redis_client.hset("item_name_to_id", item_name, item_id)
+    ```
+
+11. Delete the line with the following content:
+    ```python
+    return {"item": grocery_list[item_id]}
+    ```
+    And replace it with:
+    ```python
+    return {"item": ItemPayload(item_id=item_id, item_name=item_name, quantity=quantity)}
+    ```
+
+12.	If you would like, you can try to do a similar replacement for the other routes. Otherwise, you can just replace the entire content of the file with the lines below:
+    ```python
+    @app.get("/")
+    def home(request: Request) -> dict[str, str]:
+        url = str(request.base_url)
+        return {"message": f"Add /docs to the end of the URL to access the Swagger UI."}
+
+    # Route to add an item
+    @app.post("/items/{item_name}/{quantity}")
+    def add_item(item_name: str, quantity: int) -> dict[str, ItemPayload]:
```

Comment:
Not sure if it is rendering weird for everyone but the contents inside this function isn't showing as indented.